specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/EasyAuthConfigs.json

```diff
@@ -0,0 +1,706 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Ok",
+            "schema": {
+              "$ref": "#/definitions/AuthConfig"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create or Update Container App AuthConfig": {
+            "$ref": "./examples/AuthConfigs_CreateOrUpdate.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "ContainerAppsAuthConfigs"
+        ],
+        "summary": "Delete a Container App AuthConfig.",
+        "description": "Description for Delete a Container App AuthConfig.",
+        "operationId": "ContainerAppsAuthConfigs_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "containerAppName",
+            "in": "path",
+            "description": "Name of the Container App.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the Container App AuthConfig.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Container App deleted successfully."
+          },
+          "204": {
+            "description": "Container App AuthConfig does not exist."
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Delete Container App AuthConfig": {
+            "$ref": "./examples/AuthConfigs_Delete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AuthConfig": {
+      "description": "Configuration settings for the Azure ContainerApp Authentication / Authorization feature.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Easy Auth configuration resource specific properties",
+          "type": "object",
+          "properties": {
+            "enabled": {
```

Comment:
>enabled [](http://example.com/codeflow?start=13&length=7)

ARM recommends string enums over boolean. Generally enums make better properties than booleans. They are more descriptive, flexible and future-proof while being easier for customers to discover, understand and use. Ref: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review).

Even if you only ever want two values, you can have a string enum with True/False (or Enabled/Disabled, etc.) values which is still better than boolean, since enums allow adding values later without requiring a new API version (e.g. "Unknown", "Default", etc.), and also allow values other than just true/false.

In this case, you might consider something like status: { "Active", "Disabled" }).

Reply:
Thanks for the suggestion and detail explanation, update as you suggested.